### PR TITLE
feat(example): Improve readability of the code

### LINKED ISSUE #98: demo should be easy to read
use the standard build and standard js file instead of a full webpack example.

### PR DESCRIPTION
**Pull request by redox**
Split the example and library bundle files into several ones to improve
the readability of the example while inspecting the source code.
Fix #98 

What do you think about that one @vvo ?

### CODE REVIEW

File: example/index.html [hunk 1]
```diff
@@ -59,6 +59,10 @@
     </div>
   </div>
 
-  <script src="bundle.js"></script>
+  <!-- example dependencies -->
+  <script src="commons-chunk.js"></script>
```
Comment (bobylito) [inline example/index.html]: Maybe not the most understandable name for that file?

**Comment by bobylito** [comment]
Some lint stuff ^ :)

Seems like a handy bit of plugin :) What does it do exactly? Create separate output?

File: example/index.html (hunk 1 above)
Comment (redox) [inline example/index.html]: Yeah I tried several things but couldn't figure out the best name. Any idea?

Comment (bobylito) [inline example/index.html]: `commons`, `libs`, `libraries`?

**Comment by vvo** [comment]
I wonder if that's better. We do now see multiple files (good).

But first reaction is why "commons-chunk"? (I get it but the users?).

Then you look at the code and see:

2015-10-09-103249_1155x272_scrot

For me it's trying to show the user that the code is readable while it's not very readable and it will just raise the FUD around it.

I would no tackle this for now, it's not a big deal if we have good documentation on how to use instantsearch.js.

Users really wanting to know how the example works can just have a look at the source code.

The instantsearch demo will be the tutorial we show to users, not the example app of instantsearch.js which should be used as a `dev` tool and not something we want our users to fully understand.

The instant search demo port using instantsearch.js will be the entry point for our users to understand how to build such UI.

**Comment by redox** [comment]
> For me it's trying to show the user that the code is readable while it's not very readable and it will just raise the FUD around it.

Ideally, I would remove webpack from the `app.js` file but didn't want to remove your `require("template/...")` instructions. I understand it's not very nice, but I think it's already better than what we have. Then if we find a loader that is replacing the `require` in place without defining those `__webpack_require__` functions, maybe it's a good compromise.


> Users really wanting to know how the example works can just have a look at the source code.

I'm convinced that's a very very small part of the users.

**Comment by vvo** [comment]
I would remove the public example url, it was here only to be able to demo the current prototype easily. And stop publishing it to gh-pages also.

To our developers, users, public the integration we will show will be the instant search demo built with instantsearch.js. This will be our documentation and main example, not the example/.

The example/ was mostly done to ease the development process.

Trying to make our development example readable for the public (that do not go to github) is no more necessary if there's no running version of it.

If we try to make it somehow readable what we will endup with is more issues and FUD in our development process.

> Ideally, I would remove webpack from the app.js

Yes me too, that's why that's not what we should show to the public, but this: https://github.com/algolia/instant-search-demo/blob/feat/instantsearch.js-implementation/app.js.

I am :-1: for this PR

**Comment by redox** [comment]
> I would remove the public example url, it was here only to be able to demo the current prototype easily. And stop publishing it to gh-pages also.
> 
> To our developers, users, public the integration we will show will be the instant search demo built with instantsearch.js. This will be our documentation and main example, not the example/.

Ah if we keep that code as something only used for dev/test and we don't promote it on `gh-pages`, I'm 100% OK to close the PR and keep the previous code.

**Comment by vvo** [comment]
> Ah if we keep that code as something only used for dev/test and we don't promote it on gh-pages, I'm 100% OK to close the PR and keep the previous code.

That's it, it was a mistake to make it public without explaining the long term goal.

**Comment by vvo** [comment]
see #200

**Pull request closed without merging**